pkg/worker/worker.go

```diff
@@ -107,12 +107,14 @@ func (w *worker) Run() {
 	w.cancelToken <- true
 }
 
-func (w *worker) publishError(message pipeline.Message, err error) error {
-	errJSON, err := json.Marshal(workerError{
-		Topic:     w.mqQueue,
-		MessageID: message.ID,
-		Worker:    w.id,
-		Error:     err.Error(),
+func (w *worker) publishError(message pipeline.Message, attempt pipeline.Message, err error) error {
+	errJSON, err := json.Marshal(pipeline.PipelineError{
+		ReceivedByWorker:  message,
+		ProcessingAttempt: attempt,
```

Comment:
Could this field just be PipelineMessage? Or something along the lines of "FailedMessage"

Reply:
discussed and keeping processing_attempt for now